pybossa/view/projects.py

```diff
@@ -1343,6 +1343,26 @@ def respond(tmpl):
         if has_no_presenter(project):
             flash(gettext("Sorry, but this project is still a draft and does "
                           "not have a task presenter."), "error")
+
+        # Get locked tasks for this project.
+        locked_tasks = get_locked_tasks_project(project.id)
+        # Filter locked tasks for this user.
+        user_locked_tasks = list(filter(lambda task, u_id=user_id: task.get('user_id') == str(u_id), locked_tasks))
+        if user_locked_tasks:
+            # This user already has a locked task, take the first one being served.
+            task = user_locked_tasks[0]
```

Comment:
It does not always return the first task when there are multiple tasks locked by the worker. Instead, if returns the one with longest timeout value. 

We can refer to task scheduler code: https://github.com/bloomberg/pybossa/blob/b5d7cbb06b0e9650b03211b53ba2509837b06afc/pybossa/sched.py#L272
```
        task_id, lock_seconds = get_task_id_and_duration_for_project_user(project_id, user_id)
        if lock_seconds > 10:
            task = session.query(Task).get(task_id)
            if task:
                return [task]
```

Reply:
``` python
        remaining_time = None
        # get the locked task with the longest timeout value for the current user
        task_id, lock_seconds = get_task_id_and_duration_for_project_user(project_id, user_id)

        # if the longest timeout value is greater than 10 seconds, return the task and the timeout value, otherwise fetch another available task from db and return the project timeout 
        if lock_seconds > 10:
            # return the locked tasks and the remaining time
            remaining_time = lock_seconds

         # Set the original timeout seconds to display in the message.
         template_args['project'].original_timeout = timeout
         # Set the seconds remaining to display in the message.
         template_args['project'].timeout = remaining_time
         current_app.logger.info("User %s present task %s, remaining time %s, original timeout %s",
                                     user_id, task_id, remaining_time, timeout)
```

Reply:
Fixed via https://github.com/bloomberg/pybossa/pull/734/commits/5c71bd17e7dfa27de3c9cffc86f502acd36c1fb7
Thank you!